pkg/asset/agent/installconfig.go

```diff
@@ -120,3 +125,239 @@ func (a *OptionalInstallConfig) ClusterName() string {
+			fieldPath := field.NewPath("Platform", "VSphere", "Password")
+			logrus.Debugf(fmt.Sprintf("%s: %s is ignored", fieldPath, vspherePlatform.Password))
+		}
+		if vspherePlatform.Datacenter != "" {
+			fieldPath := field.NewPath("Platform", "VSphere", "Datacenter")
+			logrus.Debugf(fmt.Sprintf("%s: %s is ignored", fieldPath, vspherePlatform.Datacenter))
+		}
+		if vspherePlatform.DefaultDatastore != "" {
+			fieldPath := field.NewPath("Platform", "VSphere", "DefaultDatastore")
+			logrus.Debugf(fmt.Sprintf("%s: %s is ignored", fieldPath, vspherePlatform.DefaultDatastore))
+		}
+		if vspherePlatform.Folder != "" {
+			fieldPath := field.NewPath("Platform", "VSphere", "Folder")
+			logrus.Debugf(fmt.Sprintf("%s: %s is ignored", fieldPath, vspherePlatform.Folder))
+		}
+		if vspherePlatform.Cluster != "" {
+			fieldPath := field.NewPath("Platform", "VSphere", "Cluster")
+			logrus.Debugf(fmt.Sprintf("%s: %s is ignored", fieldPath, vspherePlatform.Cluster))
+		}
+		if vspherePlatform.ResourcePool != "" {
+			fieldPath := field.NewPath("Platform", "VSphere", "ResourcePool")
+			logrus.Debugf(fmt.Sprintf("%s: %s is ignored", fieldPath, vspherePlatform.ResourcePool))
+		}
+		if vspherePlatform.ClusterOSImage != "" {
+			fieldPath := field.NewPath("Platform", "VSphere", "ClusterOSImage")
+			logrus.Debugf(fmt.Sprintf("%s: %s is ignored", fieldPath, vspherePlatform.ClusterOSImage))
+		}
+		if vspherePlatform.DefaultMachinePlatform != &(vsphere.MachinePool{}) {
+			fieldPath := field.NewPath("Platform", "VSphere", "DefaultMachinePlatform")
+			logrus.Debugf(fmt.Sprintf("%s: %v is ignored", fieldPath, vspherePlatform.DefaultMachinePlatform))
+		}
+		if vspherePlatform.Network != "" {
+			fieldPath := field.NewPath("Platform", "VSphere", "Network")
+			logrus.Debugf(fmt.Sprintf("%s: %s is ignored", fieldPath, vspherePlatform.Network))
+		}
+		if vspherePlatform.DiskType != "" {
+			fieldPath := field.NewPath("Platform", "VSphere", "DiskType")
+			logrus.Debugf(fmt.Sprintf("%s: %s is ignored", fieldPath, vspherePlatform.DiskType))
+		}
+		if len(vspherePlatform.VCenters) > 1 {
+			fieldPath := field.NewPath("Platform", "VSphere", "VCenters")
+			logrus.Debugf(fmt.Sprintf("%s: %v is ignored", fieldPath, vspherePlatform.VCenters))
+		}
+		if len(vspherePlatform.FailureDomains) > 1 {
+			fieldPath := field.NewPath("Platform", "VSphere", "FailureDomains")
+			logrus.Debugf(fmt.Sprintf("%s: %v is ignored", fieldPath, vspherePlatform.FailureDomains))
+		}
+	}
+	// "External" is the default set from generic install config code
+	if installConfig.Publish != "External" {
+		fieldPath := field.NewPath("Publish")
+		logrus.Warnf(fmt.Sprintf("%s: %s is ignored", fieldPath, installConfig.Publish))
+	}
+	if installConfig.CredentialsMode != "" {
+		fieldPath := field.NewPath("CredentialsMode")
+		logrus.Warnf(fmt.Sprintf("%s: %s is ignored", fieldPath, installConfig.CredentialsMode))
+	}
+	if installConfig.BootstrapInPlace.InstallationDisk != "" {
```

Comment:
```suggestion
	if installConfig.BootstrapInPlace != nil && installConfig.BootstrapInPlace.InstallationDisk != "" {
```
This is failing unit tests due to a nil pointer deref.